### PR TITLE
feat: Add api contract to fastapi docs

### LINKED ISSUE #4720: Addition of /docs endpoint
Creation of /docs endpoint in order to have a place to view easily all the avilable endpoints with:

- a basic endpoint description.
- its development status/maturity
- requiered and optional parameters
- example of request body
- example of a succesful response
- custem errors explanation(if exist)

### PR DESCRIPTION
**Pull request by robhowley**
# What this PR does / why we need it:

* FastApi by default creates an openapi docs page which lists out endpoints
* the python feature server doesn't use pydantic models for request bodies, so the sample inputs for the endpoints aren't populated and schema unknown
* remove json request body parsing and replace it w pydantic models
* this mr does _not_ define a robust response schema for `/get-online-features`
  * simply specifies it as dict[str, any]
  * should come up w something better but as a follow up

```
feast serve --port 8080
```
generates the following at [http://localhost:8080/docs](http://localhost:8080/docs)

image

image

# Which issue(s) this PR fixes:

Fixes #4720 

# Misc

### CODE REVIEW

File: sdk/python/feast/feature_server.py
Comment (robhowley) [inline sdk/python/feast/feature_server.py:121]: fastapi is smart about types. surfaces pydantic model in the autogenerated api docs

File: sdk/python/feast/feature_server.py
Comment (robhowley) [inline sdk/python/feast/feature_server.py:74]: could (should) add validation on the model that throws if both are set. didnt do it here, bc it would be technically breaking. the code works just picks the `feature_service` value and ignore `features`. this is an unintuitive (successful) result, seems like it would make more sense to return `422` and say, "gotta pick one". maybe flag it for 1.0? @franciscojavierarceo

**Review by franciscojavierarceo — COMMENTED** [review]
Looks like we're ignoring a lot of types now. Were the type checks nontrivial to add? I've had some challenges in the past so wanted to confirm.

**Comment by robhowley** [comment]
image

im honestly unsure how this passed before. i think bc it was pulling from json it was just marking everything as dict[str, Any]? it gets an inconsistent assignment type error which makes sense since it's assigning two diff things. i can try and add asserts to please the mypy gods, but didnt want to introduce the risk of throwing where one didnt exist before (even though we know it shouldn't matter) just bc of mypy. i'll defer to you @franciscojavierarceo on this: try and add asserts or let it be.

**Comment by franciscojavierarceo** [comment]
We can add asserts and type check in subsequent PR. Thank you for this!

**Comment by robhowley** [comment]
image

flake?

File: sdk/python/feast/feature_server.py
Comment (dmartinol) [inline sdk/python/feast/feature_server.py:253]: it's not allowed, so we should fetch the `feature_view` instance as we do in the `write-to-online-store` endpoint.
let's open a GH issue for that, unless you want to fix it here

Comment (dmartinol) [reply sdk/python/feast/feature_server.py:253]: BTW: thanks for catching it!

Comment (robhowley) [reply sdk/python/feast/feature_server.py:253]: nah, not here. that should be a sep issue and `fix:` pr

Comment (dmartinol) [reply sdk/python/feast/feature_server.py:253]: https://github.com/feast-dev/feast/issues/4726

Comment (robhowley) [reply sdk/python/feast/feature_server.py:253]: i'll take care of #4726 once this pr is merged

**Comment by robhowley** [comment]
giving this a bump.

**Review by franciscojavierarceo — APPROVED** [review]

**Pull request merged**
